### PR TITLE
Add 4.x, 6.x, 7.x NAND padgen test

### PR DESCRIPTION
**Pull request by swarzesherz**
Maybe work in other versions with range scan

### CODE REVIEW

File: source/decryptor/padgen.c [hunk 1]
```diff
@@ -130,15 +130,43 @@ u32 SdPadgen()
...
-    //Just trying to make sure the data we're looking for is where we think it is.
-    //Should use some sort of memory scanning for it instead of hardcoding the address, though.
-    if (*(u32*)0x080D794C != 0x5C980)
+    //Use memory know to start scanning.
+    //If not in t Should use some sort of memory scanning for it instead of hardcoding the address, though.
+    u32 ctrStart=0, c;
+    u32 listCtrStart[] = {0x080D7CAC, 0x080D748C, 0x080D740C, 0x080D794C};
+    char* listSystem[] = {"4.x", "6.x", "7.x", "9.x"};
+    u32 lenListCtrStart = sizeof(listCtrStart)/sizeof(u32);
+    for(c=0; c < lenListCtrStart; c++)
```
Comment (archshift) [inline source/decryptor/padgen.c]: We use C99, so you can declare `c` in the for loop itself if you're not using it outside of the for loop.

File: source/decryptor/padgen.c [hunk 2]
```diff
@@ -130,15 +130,43 @@ u32 SdPadgen()
...
+        {
+            ctrStart = listCtrStart[c] + 0x30;
+            Debug("System version %s", listSystem[c]);
+            break;
+        }
+    }
+
+    //If value not in previous list start memory scanning (test range)
+    if (ctrStart == 0){
+        for(c=0x080D7FFF; c > 0x080D7000; c--)
```
Comment (archshift) [inline source/decryptor/padgen.c]: Again, declare `c` in the for loop please.

**Comment by archshift** [comment]
Could you please just change the brackets to all be on the same line as the conditional?

Other than those comments, thanks and LGTM.

**Comment by swarzesherz** [comment]
All changes done...

**Pull request merged**